### ISSUE TITLE
SFW toggle doesn't clean up the featured boards list

### ISSUE THREAD

**Issue by lluixhi** [nextchan/infinity-next#34]
It also doesn't clean up Recent Posts.

And it leaves empty blocks in catalogs.

We shouldn't have any NSFW content on the front page.

**Comment by mission712** [comment]
Oh crap. We should probably work on this immediately.

**Comment by interjection** [comment]
I think I said it before but this should probably be progressively enhanced through javascript and maybe an API.

NSFW images shouldn't load at all. The toggle is just placebo with CSS, you're still downloading animay titty images. Ideally, querystrings? Template SFW by default.

**Comment by mission712** [comment]
Maybe we could also embed the SFW status in the infinity-next cookie, and serve the user NSFW images based on that cookie? I'm not sure about how Laravel handles cookies.
Also, interjection, pls come to IRC. :3

**Comment by interjection** [comment]
@mission712
>NSFW status in session
I don't see the point in sprinkling Session::put magic, when we can store it locally in the options, and apply a querystring with the toggle using javascript.
Example:
https://nextchan.org/?nsfw
Would template NSFW, whereas it'd be templated SFW by default.

All I know is that the CSS toggle is a shit hack.

>how Laravel handles
I don't know Lara, or PHP, really. Some of it is disappointing but not enough to make me want to kill myself. PHP has definitely grown up since 2006.

>IRC
I'm idle most of the time due to work.

**Comment by mission712** [comment]
@interjection
I didn't say session, I said cookie, which is better in my opinion.

**Comment by mission712** [comment]
Actually, scrap that, the only thing IN puts as a cookie is the session cookie.

**Comment by interjection** [comment]
@mission712 Are you working on this or should I? Don't want to duplicate effort.

**Comment by interjection** [comment]
@mission712 something like this in WelcomeController:

```php
 namespace App\Http\Controllers\Content;
 
 use App\Board;
+use App\FileAttatchment;
 use App\Http\Controllers\Controller;
 use App\Http\Controllers\Board\BoardStats;
 use App\Post;
@@ -37,15 +38,25 @@ class WelcomeController extends Controller
      */
     public function getIndex()
     {
+        if (isset($_GET['nsfw'])) {
+            $worksafe = false;
+        } else {
+            $worksafe = true;
+        }
+
         if ($featured = Post::getPostFeatured()) {
             $featured->setRelation('replies', []);
         }
 
-        $featured_boards = Board::getFeatured(5);
+        $featured_boards = Board::getFeatured(5, $worksafe);
+        $recent_files = FileAttatchment::getRecentImages(30, $worksafe);
+        $recent_posts = Post::getRecentPosts(24, $worksafe);
 
         return $this->view(static::VIEW_INDEX, [
             'featured' => $featured,
             'featured_boards' => $featured_boards,
+            'recent_files' => $recent_files,
+            'recent_posts' => $recent_posts,
             'stats' => $this->boardStats(),
         ]);
     }
```

And edit all of the templates, resources/views/content/index/sections/recent_(images|posts).blade.php, and the queries in getFeatured/Recent.

I dunno if the superglobal will work for ESI, though, and don't have time to set up that kind of infrastructure.

**Comment by interjection** [comment]
@mission712 Though you might be able to pass the querystring as a parameter to esi_url().

**Comment by lluixhi** [comment]
@interjection pretty sure he's working on it right now.

**Comment by interjection** [comment]
This is fixed in 8e5f0df032af08a803cb58b14db6b59c303bb327, just needs pushed.
